src/me/remie/vulcan/leaguetasks/LeagueScript.java

```diff
@@ -109,6 +109,19 @@ private void checkTasks() {
             }, 1000, 5);
             return;
         }
+
+        // Make sure we have correct filter active.
+        final SimpleWidget dropdownMenuWidget = ctx.widgets.getWidget(657,31).getChild(4);
+        if (dropdownMenuWidget != null && !dropdownMenuWidget.isHidden() && !dropdownMenuWidget.getText().equalsIgnoreCase("all")) {
+            dropdownMenuWidget.click(0);
+            ctx.sleep(1000);
+            final SimpleWidget optionAllWidget = ctx.widgets.getWidget(657,39).getChild(1);
+            if (optionAllWidget != null && !optionAllWidget.isHidden()) {
+                optionAllWidget.click(0);
+                ctx.sleep(1000);
```

Comment:
We aren't validating that we now have the 'All' option selected. Maybe after this, we can make a check before continuing with the rest of the method. Using returns should be fine as the script will reloop this function call until the boolean value is set to true.

Reply:
The dropdownMenuWidget contains the text with the current option. 
I'm doing the check at:
`!dropdownMenuWidget.getText().equalsIgnoreCase("all")`